src/shared/behaviors/collection.html

```diff
@@ -164,19 +164,18 @@
 		_deepChanged:function(change) {
 			if (change && change.path) {
 				var path = change.path.split(".");
-				if (path.length > 1) {
-					var idx = this.data.indexOf(this.get('data.'+path[1]));
-					var model = this.data[idx];
+				if (path[1] && path[1].charAt(0) === '#') {
```

Comment:
`path.length > 1` is preferred as a bounds-checking test (or is there some other purpose being served by `path[1]` that I'm missing)